nixos/modules/services/torrent/flood.nix

```diff
@@ -0,0 +1,287 @@
+              description = lib.mdDoc ''
+                Password of Deluge RPC interface.
+              '';
+            };
+          };
+          rtorrent = {
+            host = lib.mkOption {
+              type = lib.types.nullOr lib.types.str;
+              default = null;
+              description = lib.mdDoc ''
+                Host of rTorrent's SCGI interface.
+              '';
+            };
+            port = lib.mkOption {
+              type = lib.types.nullOr lib.types.port;
+              default = null;
+              description = lib.mdDoc ''
+                Port of rTorrent's SCGI interface.
+              '';
+            };
+            socket = lib.mkOption {
+              type = lib.types.nullOr lib.types.path;
+              default = null;
+              description = lib.mdDoc ''
+                Path to rTorrent's SCGI unix socket.
+              '';
+            };
+          };
+          qbittorrent = {
+            url = lib.mkOption {
+              type = lib.types.nullOr lib.types.str;
+              default = null;
+              description = lib.mdDoc ''
+                URL to qBittorrent Web API.
+              '';
+            };
+            user = lib.mkOption {
+              type = lib.types.nullOr lib.types.str;
+              default = null;
+              description = lib.mdDoc ''
+                Username of qBittorrent Web API.
+              '';
+            };
+            pass = lib.mkOption {
+              type = lib.types.nullOr lib.types.str;
+              default = null;
+              description = lib.mdDoc ''
+                Password of qBittorrent Web API.
+              '';
+            };
+          };
+          transmission = {
+            url = lib.mkOption {
+              type = lib.types.nullOr lib.types.str;
+              default = null;
+              description = lib.mdDoc ''
+                URL to Transmission RPC interface.
+              '';
+            };
+            user = lib.mkOption {
+              type = lib.types.nullOr lib.types.str;
+              default = null;
+              description = lib.mdDoc ''
+                Username of Transmission RPC interface.
+              '';
+            };
+            pass = lib.mkOption {
+              type = lib.types.nullOr lib.types.str;
+              default = null;
+              description = lib.mdDoc ''
+                Password of Transmission RPC interface.
+              '';
+            };
+          };
+        };
+      };
+    };
+  };
+
+  config =
+    let
+      addAuthOpt = prefix: opt:
+        lib.mapAttrsToList
+          (k: v:
+            "--${prefix}${toString k} ${toString v}")
+          (lib.filterAttrs (k: v: !isNull v) opt);
+      delugeAuth = addAuthOpt "de" cfg.auth.deluge;
+      rtorrentAuth = addAuthOpt "rt" cfg.auth.rtorrent;
+      qbittorrentAuth = addAuthOpt "qb" cfg.auth.qbittorrent;
+      transmissionAuth = addAuthOpt "tr" cfg.auth.transmission;
```

Comment:
Passing in secret values through options is bad practice, as they will leak out into the nix store. So this is not really ideal. Unfortunately, Flood does not seem to have _any_ way of loading in these options decoratively without using the command line options. I am currently trying to ask on the Flood Discord if there is a current way or how a PR that addresses this would be structured.